### PR TITLE
Add UI support for RPi Pico-W

### PR DESCRIPTION
**Pull request by balloob**
~~This is a draft while we wait for the backend.~~ Requires https://github.com/esphome/esphome/pull/3284

### Wizard - Web Installation step now mentions to skip step if it's a Raspberry Pi Pico

image

### Wizard - Select device type shows Pico W as option

image

- Pick specific board also allows choosing the normal Pico
  image

### Install choose dialog - The web installation option is disabled

image

### CODE REVIEW

File: src/wizard/wizard-dialog.ts [hunk 1]
```diff
@@ -568,6 +582,9 @@ export class ESPHomeWizardDialog extends LitElement {
     } else if (this._board === "ESP32C3") {
       this._data.platform = "ESP32C3";
       this._data.board = GENERIC_ESP32C3;
+    } else if (this._board === "PI-PICO-W") {
+      this._data.platform = "RP2040";
+      this._data.board = "pico_w";
```
Comment (balloob) [inline src/wizard/wizard-dialog.ts:587]: This needs to be updated once right values are known

File: src/wizard/boards.ts [hunk 2]
```diff
@@ -20,6 +20,10 @@ export const boardSelectOptions = html`
     <option value="featheresp32">Adafruit ESP32 Feather</option>
     <option value="m5stack-core-esp32">M5Stack Core ESP32</option>
   </optgroup>
+  <optgroup label="Raspberry Pi">
+    <option value="rpi_pico">Raspberry Pi Pico</option>
+    <option value="rpi_picow">Raspberry Pi Pico W</option>
```
Comment (balloob) [inline src/wizard/boards.ts:25]: Needs to be updated once right values are known

File: src/api/configuration.ts [hunk 3]
```diff
@@ -1,6 +1,11 @@
 import { fetchApiJson, fetchApiText, streamLogs } from ".";
 
-export type SupportedPlatforms = "ESP8266" | "ESP32" | "ESP32S2" | "ESP32C3";
+export type SupportedPlatforms =
+  | "ESP8266"
+  | "ESP32"
+  | "ESP32S2"
+  | "ESP32C3"
+  | "RP2040";
```
Comment (balloob) [inline src/api/configuration.ts:8]: Needs to be updated when right value is known.

File: src/wizard/boards.ts [hunk 4]
```diff
@@ -20,6 +20,10 @@ export const boardSelectOptions = html`
     <option value="thing">Sparkfun ESP8266 Thing</option>
     <option value="thingdev">Sparkfun ESP8266 Thing - Dev Board</option>
   </optgroup>
+  <optgroup label="Raspberry Pi">
+    <option value="rpi_pico">Raspberry Pi Pico</option>
+    <option value="rpi_picow">Raspberry Pi Pico W</option>
```
Comment (balloob) [inline src/wizard/boards.ts:25]:
```suggestion
    <option value="rpipico">Raspberry Pi Pico</option>
    <option value="rpipicow">Raspberry Pi Pico W</option>
```

File: src/wizard/wizard-dialog.ts [hunk 5]
```diff
@@ -52,6 +52,7 @@ const BOARD_GENERIC_ESP32 = "esp32dev";
 const BOARD_GENERIC_ESP8266 = "esp01_1m";
 const BOARD_GENERIC_ESP32S2 = "esp32-s2-saola-1";
 const BOARD_GENERIC_ESP32C3 = "esp32-c3-devkitm-1";
+const BOARD_RPI_PICO_W = "rpi_picow";
```
Comment (balloob) [inline src/wizard/wizard-dialog.ts:55]:
```suggestion
const BOARD_RPI_PICO_W = "rpipicow";
```

**Comment by balloob** [comment]
This has been updated and verified to work. Requires https://github.com/esphome/esphome/pull/3284

**Review by jesserockz — APPROVED** [review]

**Pull request merged**